analyzers/src/SonarAnalyzer.Common/Common/AnalyzerConfiguration.cs

```diff
@@ -84,20 +85,20 @@ public void Initialize(AnalyzerOptions options)
 
             public void Initialize(AnalyzerOptions options)
             {
-                var currentSonarLintXmlPath = GetSonarLintXmlPath(options);
-                if (isInitialized && loadedSonarLintXmlPath == currentSonarLintXmlPath)
+                var currentSonarLintXml = GetSonarLintXml(options);
+                if (isInitialized && loadedSonarLintXmlPath == currentSonarLintXml?.Path)
```

Comment:
@rita-sonarsource, this caching is potentially an issues for the Rider scenario, isn't it?

Even if it is, we should handle it separately from this PR.

Reply:
It seems to be in use only for Hotspots. We could handle it on our side by passing a different folder name each time, as we discussed offline. I'm not sure it's worth opening a ticket now, since we're not expected to enable hotspots any time soon.